### DISCUSSION TITLE
What functionality do you want to see implemented or is currently lacking in wordpress_client?

### DISCUSSION THREAD

**Discussion by ArunPrakashG** [ArunPrakashG/wordpress_client#55] [Ideas]
Share your thoughts as a comment! I will try to go through each of them and check for the feasibility!

**Comment by jasalt** [comment]
Hello, I'm just familiarizing myself to the library and reading the currently supported features in readme (https://github.com/ArunPrakashG/wordpress_client?tab=readme-ov-file#-supported-rest-methods) left me wishing that CRUD actions on Post Types and Taxonomies would work fully also (at least as far as the WP API goes). 

However I assume that it wouldn't take too much extra work to do anyways, those being just behind the WP API anyways.

**Reply by ArunPrakashG** [reply]
Hi, could you share what actions in specific?
Or do you mean adding support for post types and taxonomies endpoint? In this case, I didn't have any use case for adding the support so never invested in it myself.

**Reply by jasalt** [reply]
Sure. Thinking about interfacing with WP backend having some custom types and taxonomies set, eg. WooCommerce products.

Custom post types register their own endpoints which use some generic WP REST API controller class on PHP side. I'm not deeply familiar with WP API, but I would expect that this library could pick up those also quite easily as some WP OpenAPI/Swagger generators pick them up too. I guess custom taxonomies would behave the same way but didn't look much into those yet.

Would it be currently possible to interface with custom post type endpoints via current Posts -endpoint wrapping?

**Reply by ArunPrakashG** [reply]
Yes, we have support for custom endpoints in the library. it is a bit of work to set up, though. you can check it out on the docs.
Additionally, we also have support for REST API discovery, which helps you read all endpoints on a wp site.

**Reply by jasalt** [reply]
Okay, this clarifies it a bit, thank you.
